### PR TITLE
ADEN-6957 Implement slot viewport conflicts logic

### PR DESCRIPTION
**Pull request by fraszczakszymon**

### CODE REVIEW

File: src/services/btf-blocker-service.js [hunk 1]
```diff
@@ -4,19 +4,23 @@ import { slotService } from './slot-service';
 
 const logGroup = 'btf-blocker';
 
+function disableBtf() {
+	const slots = context.get('slots');
+
+	Object.keys(slots).forEach((adSlotKey) => {
+		const adSlot = slots[adSlotKey];
+
+		if (!adSlot.aboveTheFold && this.unblockedSlots.indexOf(adSlot.getSlotName()) === -1) {
```
Comment (v3rth) [inline src/services/btf-blocker-service.js]: Did you mean `!this.unblockedSlots.includes(adSlot.getSlotName())`? :D

File: src/services/slot-service.js [hunk 2]
```diff
@@ -37,22 +40,48 @@ class SlotService {
...
+			if (distance < viewportHeight) {
+				hasConflict = true;
+			}
+
+			logger(groupName, 'hasViewportConflict', hasConflict, adSlot.getSlotName(), elementId);
+
+			return !hasConflict;
+		});
+
+		return hasConflict;
```
Comment (v3rth) [inline src/services/slot-service.js]: I think [`every`](https://github.com/Wikia/ad-engine/pull/123/files#diff-7b5430adb62b98635a56208161791dc5R53) is able to return the result so you can remove this line and maybe consider replacing `every` to `some`

File: src/services/slot-service.js [hunk 3]
```diff
@@ -37,22 +40,48 @@ class SlotService {
...
+		setState(slotName, false, status);
+	}
+
+	hasViewportConflict(adSlot) {
+		const slotHeight = adSlot.getElement().offsetHeight,
+			slotOffset = getTopOffset(adSlot.getElement()),
+			viewportHeight = window.innerHeight || document.documentElement.clientHeight || document.body.clientHeight;
+
+		let hasConflict = false;
+		adSlot.getViewportConflicts().every((elementId) => {
```
Comment (v3rth) [inline src/services/slot-service.js]: I think this function is big and separated enough to be extracted

File: src/services/btf-blocker-service.js (hunk 1 above)
Comment (fraszczakszymon) [reply src/services/btf-blocker-service.js]: This code was extracted only. You're right but it isn't supported in IE11 :(

File: src/services/slot-service.js (hunk 2 above)
Comment (fraszczakszymon) [reply src/services/slot-service.js]: Thanks! Addressed

File: src/services/slot-service.js (hunk 3 above)
Comment (fraszczakszymon) [reply src/services/slot-service.js]: Addressed 👍

**Pull request merged**